src/docs/binding-classes.md

```diff
@@ -0,0 +1,397 @@
+# Binding Classes
+
+Classes function very similarly to C# classes with some important semantic and runtime differences which affect how they are exposed in C#, but generally speaking, they can exist as a 1:1 mapping. It should be noted that while Swift provides the ability to create traditional OO code, Apple is eschewing the use of classes in their own code and preferring value types instead. It is telling that in the main Swift runtime library, libswiftCore, there are no exported classes.
+
+Swift classes can be either native Swift or Objective-C. The latter is in place strictly for compatability with legacy code and in fact, if a Swift class is declared to be `@ObjC`, it must inherit from `NSObject` and the swift compiler will generate a selector dispatcher for all methods marked as `@ObjC` but these just go to the Swift implementation, therefore Objective-C compatability is just a thin veneer on top of the Swift implementation.
+
+## Language Parity Mismatches
+Swift provides a different set of accessibility of methods that C#:
+- open - callable by all and overridable (effectively public virtual)
+- public - callable by all not overridable
+- private - callable only by members of the class
```

Comment:
I don't think private members are even exposed in the ABI file.

Reply:
I was worried about structs and just checked this example:
```swift
@frozen
public struct Foo {
    private var name: String
    public init(to: String) {
        name = to
    }
    public func Ima() -> String {
        return "I am a(n) " + name
    }
}
```
and the good news is that private members are present in the abi.json file:
```json
{
  "ABIRoot": {
    "kind": "Root",
    "name": "TopLevel",
    "printedName": "TopLevel",
    "children": [
      {
        "kind": "TypeDecl",
        "name": "Foo",
        "printedName": "Foo",
        "children": [
          {
            "kind": "Var",
            "name": "name",
            "printedName": "name",
            "children": [
              {
                "kind": "TypeNominal",
                "name": "String",
                "printedName": "Swift.String",
                "usr": "s:SS"
              }
            ],
            "declKind": "Var",
            "usr": "s:20GenericTestFramework3FooV4name33_30D99E5814F18B95AFD2006B0E3423D7LLSSvp",
            "mangledName": "$s20GenericTestFramework3FooV4name33_30D99E5814F18B95AFD2006B0E3423D7LLSSvp",
            "moduleName": "GenericTestFramework",
            "isInternal": true,
            "declAttributes": [
              "HasStorage",
              "AccessControl"
            ],
            "fixedbinaryorder": 0,
            "hasStorage": true
          },
     ... extra removed
```